site/packages/aa-core/provider/extend.md

```diff
@@ -0,0 +1,53 @@
+---
+outline: deep
+head:
+  - - meta
+    - property: og:title
+      content: extend
+  - - meta
+    - name: description
+      content: Overview of the extend method on ISmartAccountProvider
+  - - meta
+    - property: og:description
+      content: Overview of the extend method on ISmartAccountProvider
+---
+
+# extend
+
+Allows you to add additional functionality and utility methods to this provider via a decorator pattern.
+
+## Usage
+
+::: code-group
+
+```ts [example.ts]
+import { provider } from "./provider";
+
+const extendedProvider = provider.extend((provider) => ({
+  debugSendUserOperation: (...args) => {
+    console.log("debugging send user operation");
+    return provider.sendUserOperation(...args);
+  },
+}));
+
+extendedProvider.debugSendUserOperation(...);
+```
+
+<<< @/snippets/provider.ts
+:::
+
+## Returns
+
+### `this & R`
+
+Where `R` is the return type of the decorator function.
+
+::: warning
+The extend method will not allow you to override existing methods or properties on the provider. To do this, you'll have to use class inheritance.
+:::
```

Comment:
we should put this in code comment as well.

Reply:
good call, updated!